keystore/v2/keystore/filesystem/backend/utils.go

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2020, Cossack Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package backend
+
+import (
+	"bytes"
+
+	"github.com/cossacklabs/acra/keystore/v2/keystore/filesystem/backend/api"
+)
+
+// SplitPath splits a path into components by separator.
+func SplitPath(path string) []string {
```

Comment:
> can we replace `SplitPath` with `filepath.Split` or `strings.Split`?
>
> [is `JoinPath`] duplicate of `filepath.Join`?

These functions were motivated by the following considerations:

- Since “key paths” are not file paths we should not use `os.PathSeparator` for them. Instead we’d better define _our_ path separator as always `/`. However, writing `strings.Split(path, "/")` seems like unnecessary hardcoding, `strings.Split(path, backend.PathSeparator)` looks too verbose, so I’ve just made these two utilities.

- Another realization that I’ve had is that `strings.Split()` works on _strings_ (as in UTF-8 encoded Unicode text). However, in offline discussions I remember some ideas about using binary data as client or zone IDs. While Go allows to keeps arbitrary bytes in `string`, all those string utilities expect actual text. In order to get correct behavior we should use `bytes.Split` instead. Having separate utilities which are consistently used makes it easier to choose them and not choose `bytes` when you need `strings` or vice versa.

However, now that you’ve mentioned it, we could probably do like Windows does: have some canonical path separator but accept either of them. It just boils down to putting every paths through some

```go
var pathSeparators = strings.NewReplacer(
        "/", os.PathSeparator,
        "\\", os.PathSeparator,
)

// when you need to pass a path to OS:
os.Open(pathSeparators.Replace(path))
```

And then just let the code use `filepath.Join()` if they want, or whatever. We’d need to take special care _if_ we us binary data in paths, but maybe accepting `string` instead of `[]byte` is a good enough reminder about that.

Reply:
can you comment it in the source code? because I am sure that I will forget your explanation after some time) and for other future contributors, to prevent removing it)